Modules/_threadmodule.c

```diff
@@ -1022,6 +1022,13 @@ rlock_traverse(PyObject *self, visitproc visit, void *arg)
     return 0;
 }
 
+/*
+helper function used by rlock_locked and rlock_repr.
+*/
+static int
+rlock_locked_impl(rlockobject *self) {
```

Comment:
You should use it in rlock_repr().

Reply:
> I would prefer to have the same logic in locked() and repr(). repr() checks `self->lock.thread`. Can we do the same here?

I am wondering if we check `self->lock.thread`means that the lock is really **locked** ? 
Call to the `PyMutex_IsLocked`API seems to me more appropriate. But perharps, I missed something

Reply:
I also prefer calling `PyMutex_IsLocked()`. It seems to be "more correct" :-)